### ISSUE TITLE
Convert a Benbox engraver to GRBL

### ISSUE THREAD

**Issue by Rick23n** [arkypita/LaserGRBL#436]
I have a chinese 16w Laser Engraver which came with Benbox to install which worked initially with NANO(324p)/LX-Nano with a CH340 chip but now the command buttons will only move the Y axis one way or activate the laser on full power.I have now installed LaserGRBL v2.8.52 which seemed to install correctly but when I import a file to print I cannot get the 'connect' button to activate. It tries to connect but then shows "disconnected" and the Session log reads:
5/09/2018 6:54:45 PM	SetStatus   	Machine status [Connecting]
5/09/2018 6:54:45 PM	OpenCom     	Open COM4 @ 115200 baud Ctrl-X
5/09/2018 6:54:45 PM	SendImmediate	Send Immediate Command [0x18]
5/09/2018 6:54:49 PM	SendImmediate	Send Immediate Command [0x18]
5/09/2018 6:54:55 PM	OpenCom     	Connection timeout!
5/09/2018 6:54:55 PM	CloseCom    	Close COM4 [CORE]
5/09/2018 6:54:55 PM	SetStatus   	Machine status [Disconnected]
5/09/2018 6:55:24 PM	Program     	------------ PROGRAM STOP ------------
Please help

**Comment by gisep** [comment]
I bought a similar one and when trying to run it from LaserGRBL I had problems. With grbl firmware the motors became hot and it didn't run well, then when tried to put original firmware to run again the chinese software benbox i didn't find the firmware on the net and the seller told me it wasn't opensource.
I finally bought an arduino UNO with cnc shield, that's a lot better; the old shield neither had PWM.

**Comment by Rick23n** [comment]
I reinstalled Benbox too and now all the controls are messed up. The Y axis will only move one way and the burn button will only burn in one continuous line and the only way to stop it is to turn off the power to the laser!

**Comment by Rick23n** [comment]
I have flashed my chinese Nano328 with the 1.1GRBL and all works fine (would NOT flash with version 2.8) My question now is can I upgrade to V2.8.35 and still have functionality of the Arduino Nano328? My reason to upgrade is the laser will not engrave the first (or bottom) line of my image. I thought the upgrade might overcome this as suggested in post 22nd Dec 17 by arkypita.

**Comment by StuartB4** [comment]
This is from the FAQ page for LaserGRBL.

> LaserGRBL work with any engraver that use Grbl as controller. This also include a lot of mounting kit from ebay, bangood, aliexpress, amazon, based on arduino uno/nano or other customized boards with ATmega328P chip.
> 
> Often some of these markers do not use Grbl but they use other control firmware/protocols (benbox or others) but is it possible to reprogram their firmware to use Grbl. You must google about how to do that with your specific controller.

So you should be able to upgrade to LaserGRBL 2.8.52, the latest and it should work.
Providing your GRBL settings are correct.

**Comment by Rick23n** [comment]
10th Sept. Laser is now working with 2.8.52 but I do still have an issue when engraving 'horizontal'. It will only engrave the vertical lines. I am thinking it may be in the GBRL configuration setting.; Its ok if set to 'diagonal' but unfortunately this changes the size of the image when it's engraved (frustrating). Vertical is not an option either because then it misses all the vertical lines. I thought it also might be speed and power but if I set any higher, then any scrolls or words are burnt too much. My speed setting is 2000 and power 200 which engraves correctly except for the horizontal lines. Any suggestions????

**Comment by StuartB4** [comment]
Can you paste a copy of your GRBL settings.
Just goto GRBL menu in LaserGRBL and click Grbl Configuration. That should show the list of settings.
Stretch the window down so it shows all the settings.
Press Print Scrn on your computer and then open MS Paint and click Paste.
You can draw a box just round the settings part of the image and Cut it, then click New. 
Click Dont Save, then click Paste and Save the image and post it here by clicking on "selecting them" at the 
bottom of this box.

Like this:
![untitled2](https://user-images.githubusercontent.com/16980026/45298598-e7e56c80-b500-11e8-9087-0cf8482c6a99.jpg)

**Comment by Rick23n** [comment]
![grbl config 0- 121](https://user-images.githubusercontent.com/42868495/45352051-e13b2000-b5fa-11e8-9dce-a721a54c924c.jpg) StuartB4 -  if you can view this file you will see that it is missing $122=10.000 m/sec^2, $130=230mm,$131=285mm,$132=200mm(I only have X & Y at the moment) This is because my laptop screen isnt big enough to show all at once. I do not have MS Paint and my anti-virus will not allow me to download it so I have done the conversion in Paint.Net

![image](https://user-images.githubusercontent.com/42868495/45353449-2c0a6700-b5fe-11e8-8bff-cb9585e57f4e.png)This image shows what is happening to my engraved image.

![image](https://user-images.githubusercontent.com/42868495/45354164-670d9a00-b600-11e8-9b18-b0ef7ce43e1e.png)This image shows the setting I am using to send to the Arduino Nano328p

Thank you for your interest I really appreciate it

**Comment by StuartB4** [comment]
You should upgrade GRBL to version 1.1, then you have an extra setting $31 which is Laser mode.

> The main difference between default Grbl operation and the laser mode is how the spindle/laser output is controlled with motions involved. Every time a spindle state M3 M4 M5 or spindle speed Sxxx is altered, Grbl would come to a stop, allow the spindle to change, and then continue. This is the normal operating procedure for a milling machine spindle. It needs time to change speeds.
> 
> However, if a laser starts and stops like this for every spindle change, this leads to scorching and uneven cutting/engraving! Grbl's new laser mode prevents unnecessary stops whenever possible and adds a new dynamic laser power mode that automagically scales power based on current speed related to programmed rate. So, you can get super clean and crisp results, even on a low-acceleration machine!
> 
> Enabling or disabling Grbl's laser mode is easy. Just alter the $32 Grbl setting.
> 
>     To Enable: Send Grbl a $32=1 command.
>     To Disable: Send Grbl a $32=0 command.

You have $110 set to 1000 maximum in GRBL but you have set the engraving speed to 2000 in the Target image settings.
Thats why it has the red box around it, it wont go 2000 when it's set to 1000 in GRBL, you need to change it to 2000 in GRBL.

**Comment by Rick23n** [comment]
Thank you for getting back so quickly and yes I have Windows 10 and yes I found paint. 
LaserGRBL v2.8.52 is what I'm using to create the code for the Arduino so I thought that grbl v1.1f was a much earlier version. My mistake. When I downloaded grbl v1.1f it doesn't contain a Windows installer exe file. How and where do I find this. (I am a complete novice at this as you must have guessed by now) OR is this the version I need to flash to the ArduinoNano328p?? I made the assumption that the program that generates the g code was the one that contains the settings $30, $31 & $32. I am confused !

**Comment by StuartB4** [comment]
Yeah thats correct, lasergrbl 2.8.52.
What you need to do is download GRBL 1.1f from here:
https://github.com/gnea/grbl/releases
You can just download the .Hex file

Then you download Xloader from here:
https://my.pcloud.com/publink/show?code=XZvPL17ZwQHksJSJSvRKcR4vzvGIwm1jUSQV
It's uploaded to my PCloud account.

Extract the XLoader file to anywhere you like.
When you have both run XLoader.exe from the folder you extracted it to and connect the Arduino to the correct com port, select nano from the dropdown list and set the correct baud rate, should be 115200, browse to where the .hex file is and click Upload.
Thats it.

Like this:
![untitled2](https://user-images.githubusercontent.com/16980026/45375449-e09b8d00-b5ec-11e8-83e8-fb512f2a95a9.jpg)

Then you can edit the GRBL settings from inside LaserGRBL in the list of settings in the image you posted.

**Comment by Rick23n** [comment]
Thank you!!! My Laser is now up and running with the new settings added. It would not connect initially so I had to experiment with the baud rate (is 115200) and also the correct Arduino (NANO(324p)/LX-Nano) The image is crisp and it is engraving the first line, a little lighter than the rest but I guess that will be in the speed and laser power settings.
Again I can't thank you enough for your time. I will now switch my attention to building a "Z" axis as my setup does not allow for changing material thickness. I hope there is a build out there that will enable me to use grbl settings for this.

**Comment by Rick23n** [comment]
I am still having a problem with the first horizontal line. Have tried different speed and power setting to no avail. The line is still faint compared to the verticals

**Comment by StuartB4** [comment]
Hi Rick.
You could try changing this setting in the Settings on the GRBL menu.
Just to see if it makes a difference.
If it's enabled disable it or vice versa.
![untitled](https://user-images.githubusercontent.com/16980026/45441852-2d986580-b6b8-11e8-9e5f-d79ddc6df6ed.jpg)

**Comment by StuartB4** [comment]
Can you upload the original image you used here so I can try it on my machine.

**Comment by StuartB4** [comment]
Just out of interest, did you enable Laser Mode in the GRBL settings, if not set $32 to 1 instead of 0.
Also set $30 to 255 (maximum laser power) and $31 to 0 (minimum laser power).
Then change these 2 values to M4 and 255.

**Comment by Rick23n** [comment]
Hi Stuart
Tried the MultiDirectional setting - no change,yes I have the Laser Mode enabled also $30 = **9000** and $31 = 0. M4 did not make any change. I don't have the "optical correct" option, was it supposed to be included in grbl1.1f?? So I still have only 25% line image for the first line as compared to 100% for the vertical lines, the border vertical are darker than the internal vertical lines(it appears that the laser burns twice on the change of direction) $110 & $111, I have left at 500 This gives a projected time of approx 1hr. Using "Diagonal"the projected time is about 2 hrs
If I use the "vertical" line to line option, then the vertical lines are affected. (i.e. 25% only) I have left the line quality at 11 L/mm
![image](https://user-images.githubusercontent.com/42868495/45477767-9dc0ed00-b785-11e8-9158-38adcf4f2278.png)
This is the image I have created in MS Publisher, saved as a .png file to import to Lasergrbl.
I can get a reasonable image with diagonal set to Pwr2000 Speed 500 and one direction lasering , If I use Multidirection it burns the perimeter lines, and YES I had to restart grbl to activate Multidirection.
Just as a matter of interest Stuart, where are you located? I am on the Gold Coast Queensland

**Comment by arkypita** [comment]
Can you also post a picture of the result, to better understand the issue?

**Comment by arkypita** [comment]
For this kind of job I suggest vectorize + filling or vectorize only

**Comment by fantilator** [comment]
@Rick23n 
Can you make you program again as on the photo where it failed.
Then save the program.
open the program in notepad or something and copy all the G-code in www.NCviewer.com
after this press the <plot> button and examing the drawing on errors.

When I did this with the drawing provided above I see this.

NCviewer:
![image](https://user-images.githubusercontent.com/23620190/45481411-bf9d7e00-b74b-11e8-8c75-4d7e1b691ce4.png)
LaserGRBL:
![image](https://user-images.githubusercontent.com/23620190/45481709-86194280-b74c-11e8-8520-7d0470e65ca5.png)

Here the bottom line is indeed not projected, maybe it's to close to the edge.
Try to lift it up a bit in the image.

**Comment by StuartB4** [comment]
Hi Rick.
""OpticalCorrect" is in the "Grayscale" drop down box and is part of the Latest LaserGRBL 2.8.52 not GRBL.
It should be in the "Import Raster Image" settings.

**Comment by Rick23n** [comment]
The best result yet has been : 
![image](https://user-images.githubusercontent.com/42868495/45492767-821f0c00-b7b0-11e8-9971-962e5ee030e2.png)
![image](https://user-images.githubusercontent.com/42868495/45492952-ee9a0b00-b7b0-11e8-9e9a-287acee02881.png)
![image](https://user-images.githubusercontent.com/42868495/45493741-c9a69780-b7b2-11e8-8870-b2a0195df60a.png)
The best I could do with the G Code:
![image](https://user-images.githubusercontent.com/42868495/45493930-48033980-b7b3-11e8-8b39-afbb47677bd6.png)
No Stuart  as you can see I dont have the 'Greyscale" drop down box  and I  re downloaded LaserGRBL 2.8.52 tonight. I have tried to shift the image up but it made no difference. The answer seems to be in getting the Laser power and speed right. Tried "Vectorizing" but only lost lines altogether.
Any thoughts are greatly appreciated

**Comment by StuartB4** [comment]
> No Stuart as you can see I dont have the 'Greyscale" drop down box

If you go to File, Open and load an image the "Import Raster Image" box should open.
Grayscale drop down box should be there.

I also noticed your $110 and $111 in GRBL settings are set to 250, but you have the Engraving speed in the Target Image settings box set to 500.

![untitled2](https://user-images.githubusercontent.com/16980026/45497557-07ce9780-b770-11e8-8627-f32b7d536297.jpg)

It wont engrave at 500 speed if you have it set lower in GRBL.
Thats why theres a red box around the Engraving Speed box, to let you know something is not right.
You can set $110 and $111 in GRBL Settings to a higher number, up to the maximum speed your machine can move.
As long as you dont exceed that number in the Target Image Engraving Speed box it will be fine
and the red warning box will not appear.

![untitled](https://user-images.githubusercontent.com/16980026/45497586-1ddc5800-b770-11e8-94e0-f9eb82bbff39.jpg)

**Comment by StuartB4** [comment]
$30 should be set to maximum 255, that is the maximum power your laser will produce using from the  PWM signal from the Arduino. It's a 0 to 5volt signal,  0 is lowest power and 5volts is maximum (255)
0 is Lowest 255 is Full power, you don't need numbers in their 1000's.

**Comment by StuartB4** [comment]
To see the GCode properly as a text file:
Load an image and get it ready for engraving but dont actually start the engraving process.
Goto File, Save Program.
Give it a name and save it.
Then you can open it in a text editor to view it or edit it as you wish.

**Comment by StuartB4** [comment]
Hi Rick. Just a thought, have you tried the same image on a different type of wood? 
All materials react differently to laser and I know some types burn better than others.

**Comment by Rick23n** [comment]
Thanks Stuart. The wood is ok. I have partially completed it (I stuffed up) and the image is fine except for the first horizontal line showing approx 50% lighter than the rest of the burn. With regard to the laser power, I saw somewhere on a forum it shows Laser Power 0-255 OR 0-12000 so when I wasn't having any luck I tried 12000 and got a better result but now I have it set to 255 and the result is 'becoming' acceptable. I still have to tweek something to get a crisp image. Thanks for the tip on File saving and yes I did get the engraving speed settings mixed up.
This is a screen shot where I can't see the Greyscale Setting
![image](https://user-images.githubusercontent.com/42868495/45585138-efeb4500-b922-11e8-822d-0cfc8cfe528c.png)
The other thing that I have a problem with is Image Size. I dont know how to control the 'vertical' size after I input the 'horizontal' measurement. Even then the figures do not correspond to the actual size.
To get 93.5mm x 274.5mm I have to input 30 x (and it automatically comes up with 90 as the vertical measurement) and it does show size in mm. And the offset I have to input 1mm to get approx 5mm. 
Any thoughts?

**Comment by StuartB4** [comment]
Hi Rick.
Dont know why you dont get the Grayscale option when it's the same version as mine, you would have to ask the Arkypita, the developer why. You can just open a new issue about that, he may see it here but may not.

Yeah about the laser power settings. Some people have a 15 Watt laser which is 15,000 milliwatts.
So they set the $30 to 15,000 in GRBL, then when they change the setting in LaserGRBL to say 800 they see it as being 800 milliwatts,  or a setting of 7500 they see as 7500 milliwatts or 7.5 Watts, up to the maximum 15,000 /15 Watts. 
Setting $30 to 255 in GRBL and then setting it to 128 in LaserGRBL is the same thing, 128 would be just about equivalent to half the laser power or 7.5 Watts. 64 would be a quater power and so on.
If you were using a router spindle instead of the laser I think the setting would be for the spindle motor speed, 15000 rpm.

As for the image size problem I'm not sure what you mean by controlling the Vertical size.
Do you mean the Width and Height settings in the Target Image settings?
If you load an image say 200 by 300mm's it should show you that size in the Target image settings W and H boxes. W 200 and H 300.
If you change the Width setting the Height setting should change automatically to keep the image ratio correct.
The only problem you should get there is if you enter a W or H size which is larger than the GRBL settings for $130 and $131. 
If they were set for 150 and 200 and you try to enter 155 in the W or 205 in the H box you would get a Red box around the W or H setting box so you know there is a problem.

![untitled](https://user-images.githubusercontent.com/16980026/45586262-cbc04180-b8eb-11e8-9448-5146322240e4.jpg)

**Comment by Rick23n** [comment]
Hi Stuart, my GRBL setting $131 is 230mmW and $132 is 285mmH As shown on the Target Image, to be able to burn the correct size I need to enter 30mm to get an image width of 97.5mm ???? I cannot see any conversion setting to be able to adjust this.
My laser is 16watt which should give me an average 8/11 watts.
![image](https://user-images.githubusercontent.com/42868495/45586389-1e284f00-b93a-11e8-95ee-46999d01718b.png)
I hope this makes it clearer.

**Comment by StuartB4** [comment]
> Hi Stuart, my GRBL setting $131 is 230mmW and $132 is 285mmH

I think you mean 130 and 131, X, 132 is Z axis.

Just realised why you have no Grayscale settings box, it only to appears if you load a colour image like a photo in.

If the image you load in is 30 x 90 then it should show them numbers in the W and H boxes.

This image is 38mm's X 38mm's and thats what shows in the W and H boxes, as it should.

![39mm center image2](https://user-images.githubusercontent.com/16980026/45586997-404cad80-b8f7-11e8-954b-9549a66ae787.png)

![square](https://user-images.githubusercontent.com/16980026/45587011-75f19680-b8f7-11e8-9025-ec6e1eda7486.jpg)

If I change number in the W the H changes to match.

![square2](https://user-images.githubusercontent.com/16980026/45587028-c668f400-b8f7-11e8-86bd-bd5962e42137.jpg)

Does the image you load in have a white border around it, or is the image right up to all 4 edges. 
If it has a border round it you can use the crop tool to just grab the image.
Just click the 6th icon along the bottom (Crop tool).
Draw a box around the actual image with the left mouse button held down then release it.
Click Next and see what the W and H numbers are.

![crop](https://user-images.githubusercontent.com/16980026/45587083-78082500-b8f8-11e8-97ed-2af3364a3636.jpg)

**Comment by Rick23n** [comment]
No border ,I made sure of that. To explain in more detail ; if I input 30mmW in the Target Image Size then when I create it, it shows 31mmW x 88mmH and that will produce an image size approx what I need. If I input 93.5mmW( which is the actual size of my imported image)93.5W x 274.5H) it will then it will try to burn an image approx 270mmW x 816mmH which is outside the constraints of the workspace $130 & $131 of 230W & 285H. Is it possible the problem is in the imported image from MS Publisher. Maybe if I learn a little more about Inkscape it may sort it out.

**Comment by Rick23n** [comment]
I also see that M3 Laser Mode should allow me to focus the Laser. I cannot turn the laser on manually. The only time the laser is activated is when I start the Burn program. Have I missed something or is it the chinese version of the Arduino that doesn't support this. There is a switch on the laser but it appears to do nothing even with the laser in burn mode. Maybe the answer is to put a switch in from the 5v+ to the laser on very low power for focusing.

**Comment by fantilator** [comment]
Try typing  M3 S100 and see what the laser does. M5 stops the laser.

**Comment by Rick23n** [comment]
I'm sorry I dont understand. Type M3 S100 where? In LaserGRBL I only have the option to select M3 or M4.
 I dont have anywhere to input a command apart from changing the $ commands in GRBL Config.

**Comment by Rick23n** [comment]
Ok , connect the laser and then I will have access to the command line!

**Comment by Rick23n** [comment]
I dont seem to be able to adjust the laser power. Set at 5 or 255 is same. Yes I have changed the grbl settings $30 to match. With or without Laser Mode $32 enabled. Also with PWM enabled. ??

**Comment by Rick23n** [comment]
Yes 3 pin connection

**Comment by Rick23n** [comment]
Thanks Stuart I will work on that

**Comment by Rick23n** [comment]
Set up the button all works except for Laser power still 100% full on

**Comment by StuartB4** [comment]
You say
>  There is a switch on the laser but it appears to do nothing even with the laser in burn mode.

Are there any marks or is there any writing on the board close to the switch.
I know some of them lasers have a switch for TTL/PWM mode or Low Power mode which allows you too focus and then press it again for Hi Power or burning mode, depending on if the switch is on or off.
Is it a latching switch or just a momentery action, does it lock in to one position when you press it and unlock when you press it again?

**Comment by StuartB4** [comment]
Oh ok. 
Seems strange that you can't vary the laser power.

Have you tried loading a colour image in to LaserGBL?
If not, can you try it.
Just load any colour jpeg image.
Set the Brightness and Contrast to suitable levels.
Select Line To Line Tracing.
Set amount of lines to 8 or whatever you want. More lines=higher quality but slower longer burn time.
Click Next.

![untitled2](https://user-images.githubusercontent.com/16980026/45598344-5aa78980-b9d2-11e8-9898-5c530045dc7e.jpg)

Then in the Target Image settings just make the W something like 100, the H should change to the correct size.
My settings for the laser are: 
Engraving Speed: 1000
Laser on: M4
Laser Off: M5
S-Min : 0
S-Max: 255
Click the Create button.
Then burn the image on to whatever material you want/have. Not pure white though, try something a bit off white.
It should do a greyscale engraving.

**Comment by fantilator** [comment]
@Rick23n 
Read this post and revert here if this is the same problem as yours.
Please also post brand, type and picture of the laser.

**Comment by arkypita** [comment]
> Dont know why you dont get the Grayscale option when it's the same version as mine, you would have to ask Arkypita, the developer why.

The grayscale conversion option is available only if you load a colored image. If the image is already a greyscale or bw no conversion option is shown.

![image](https://user-images.githubusercontent.com/8782035/45599747-78d0b200-b9f1-11e8-90b7-d28448a26d4c.png)

**Comment by arkypita** [comment]
@StuartB4 

> Just realised why you have no Grayscale settings box, it only to appears if you load a colour image like a photo in. 

You have got it!

More in detail: lasergrbl "detect" if the image "look like" grayscale by sampling some pixels of the image. If all the sampled pixels are gray (same R-G-B values less or more, with some tollerance) the image is considered as grayscale and no color conversion option are shown.

**Comment by Rick23n** [comment]
![image](https://user-images.githubusercontent.com/42868495/45602338-cab02100-ba5f-11e8-90e3-06e62744e9ff.png)
![image](https://user-images.githubusercontent.com/42868495/45602353-0b0f9f00-ba60-11e8-945b-86bb6a5fcad8.png)
![image](https://user-images.githubusercontent.com/42868495/45602365-37c3b680-ba60-11e8-9ab3-cf8120c7111e.png)
These are the photos of my Laser and control board containing the Arduino(NANO(324p)/LX-Nano) 
First to answer your question Arkypita ; the first horizontal line, the laser fires but it only burns at 25% of full power. I have set up a 5mm inset befor it starts.

**Comment by fantilator** [comment]
@Rick23n 
First thing I spot, put switch on TTL mode.
Second: you TTL- (open hole next to the yellow wire) is not connected, does your manual say something about connecting the TTL-?

try to do following things please.
1) connect laser in software, type in "M3 S0" command
2) What does the laser (normally the fan should run, but laser is off (because you run it at 0%)
3) pull out yellow cable, while laser is running (TTL) see what this does. (2 options, laser stays off, laser shoots on)
4) turn machine completely off
5) replug cable back
6) connect TTL- (empty connection next to yellow cable) to your black wire of the power supply. (just strip 2 end of a wire and prone them in) make sure you do not short circuit with the red wire.
7) turn machine back on and connect.
8) type M3 S100 and see what it does.

If you should see or hear any unusual things switch machine off right away.

I also see that if laser mode is enabled, you should use M4 Sx instead of S3. with X a value between 0 and 255
The usage of the laser within g-code is simple:

M3 turns the laser on (with constant power),
M4 turns the laser on with adaptive power,
M5 turns the laser off.
S1 - S255 sets the power of the laser (0 - 100%)

**Comment by Rick23n** [comment]
The "Manual" is very basic and mostly in Chinese. What little English there was isn't helpful so to get this far I've relied on people such as yourselves and am very appreciative of your help.
Something interesting - I experimented with the Laser power with the M4 Sxx command and found that to turn the Laser ON but NOT burn my final setting was M4 S.0125 so Stuart I have set this figure in the "Laser ON/OFF" button but it does open up yet another road to follow.

**Comment by Rick23n** [comment]
fantilator - I did reply to you but it is not showing- I will repeat - Command M3 S0 does NOT turn the laser ON but when I remove the yellow TTL cable, the laser comes ON full power. I have found that I CAN adjust the power but with strange settings S .0125 +++. I am not sure about grounding the pin beside the TTL yet but may do if all else fails. Yes I am using M4 with the laser mode now.

**Comment by arkypita** [comment]
- In my opinion grounding of TTL connector is not needed because the TTL ground is shared with power ground.

- If detaching TTL result in "ON FULL power" seem that your driver has an inverted logic (remember what I say some post ago about TTL control: 5V should be LaserON, 0V should be LaserOFF)

Did you have a multimeter? Can you measure voltage between GND and TTL pin and check that it vary with the S value?
- M3 S12000 (should be 5V)
- M3 S6000 (should be about 2.5V)
- M3 S0 (should be 0V)

If you don't have a multimeter, could you put a LED with 220ohm resistor series, between TTL and ground and see what append to its intensity? S12000 should be bright, S6000 half bright etc.

**Comment by arkypita** [comment]
**More:** As I previously say "The TTL pin is on pin D11" This is the GRBL standard. I don't know nothing about benbox but maybe the pin is not the same. If grbl control the pin D11 but the TTL cable is connected on another pin the problem would be explained.

It is not possible to see it by the picture, you should check it using multimeter.

Disconnect your pwm connector from the driver side, put multimeter in "beep" continuity function, insert one tip inside pwm connector and with the other tip test each arduino soldering.

Only D11 should "beep" when probed

**Comment by Rick23n** [comment]
Arkypita thank you for your interest. Yes only D11 shows continuity on TTL. At first at M3 S12000 when I measured the voltage there was no change at S6000. Then I started at S1 and these are the figures it showed on TTL ;
S1 = 38.2mv
S2 = 56.6mv
S5 = 111.7mv
S10 = .203v
S50 = .936v
S75 = 1.395v
S100 = 1.85v
S200 = 3.69v
S255 =4.7v
S1000 = 4.7v
S12000 = 4.7v
When I engraved an image and set Sxxxx ; Protocol Settings Buffered / Fast , Raster Unidirectional Engraving (the first voltage is on the "away" pass the second is on the "return" pass.
S2 = 2.03/2.36v
S10 = 2.03/2.8v
S100 = 2.03/3.09v
S255 = 2.03/3.13v
If I set  single burn (uncheck Unidirectional and restart GRBL ) then the voltage shows 2.03v on the first pass and 3.13v on the second pass.
I hope I have explained this clearly.

**Comment by arkypita** [comment]
> Arkypita thank you for your interest. Yes only D11 shows continuity on TTL. At first at M3 S12000 when I measured the voltage there was no change at S6000. Then I started at S1 and these are the figures it showed on TTL ;
> S1 = 38.2mv
> S2 = 56.6mv
> S5 = 111.7mv
> S10 = .203v
> S50 = .936v
> S75 = 1.395v
> S100 = 1.85v
> S200 = 3.69v
> S255 =4.7v
 
For what you say, look like that you modulate power in a range of S0-S255, not S0-S12000 as you say some post ago.

![image](https://user-images.githubusercontent.com/8782035/45688985-12fc3b80-bb53-11e8-8616-fb9b673dbc45.png)

Do you have changed those values to 0-255, correct?

**Comment by Rick23n** [comment]
Yes, in all cases I set the power to Sxxx in "Target Image" and also in grbl Config $30 with Laser Mode enabled changing from S0 up to S255. When I used setting above S255 it did not change from 2.03/3.13 Volts. on the TTL line. When I posted before using 0-12000 I was not sure of what I was doing and it seemed that I got a better burn at 12000 but now i am not so sure because 255 IS at full power (4.7v)

**Comment by Rick23n** [comment]
Thank you to the above for your support. I think I have it all sorted out now except for the image size that is shown in Laser grbl in mm actually prints 3x the size but I can work with that unless there is an obvious setting that I can adjust.

**Comment by StuartB4** [comment]
Don't know why it would burn bigger than it's supposed to.
Can you check your stepper motor movements are set correctly.

Easy way to do it is to open LaserGRBL and Connect.
Left Click the Right slider once then use the Up-Down keys on your keyboard to set it to 20 shown in the picture below.
![untitled](https://user-images.githubusercontent.com/16980026/46077980-a6a0be00-c18a-11e8-899c-3871e92fdf03.jpg)

Move the laser to about the middle of the X Axis rail.
Make a mark on the rail as close as possible to the laser on the left side of where the laser stopped. 
Click LaserGRBL Right arrow key once to move the laser 20mm's.
When it stops measure the distance it moved with a ruler from the mark on the rail
to the left side of the laser.
If the measurement is 20mm's then the stepper is set correctly.

If it's different to 20mm's you have to change the $100 and $101 settings.
Do the same for the Y Axis.

**Comment by StuartB4** [comment]
Ok Rick.

**Comment by Rick23n** [comment]
Settings $100 = 77 & $101 = 80.5 now is perfect thank you

**Comment by arkypita** [comment]
Very happy of your resut.
Thanks to @StuartB4  for big effort giving support.

Will be nice if you want to draw a step-by-step guide to help other people converting a Benbox machine to a GRBL one. I think benbox is very limited when grbl has a lot more of support, UI programs etc

**Comment by Rick23n** [comment]
I will try to explain to the best of my ability, but it may take some time to get it right. How should it be posted? On the forum? or is there somewhere else? I dont think Utube would be something that I could do.

**Comment by StuartB4** [comment]
Hi Rick.
If you do decide to do a Tutorial / Guide, you might find a bit more info here that may help.
[Benbox Forum](http://benboxlaser.us/index.php)

**Comment by Rick23n** [comment]
Thank you Stuart. I'm not sure if I will be able to yet, I am still learning. When I have it all sorted out I will certainly give more thought to sharing my experiences with everyone. I must admit the 16000mw laser is not living up to expectations. I would have thought cutting 3mm thick MDF (composite) board would have been an easy asignment but after full power at 50mm/min and 8 passes it leaves me puzzled

**Comment by StuartB4** [comment]
Hi Rick.
I noticed from the picture you posted of the laser control board with the arduino nano on it
that it has the Z Axis plug in.
Does your Z axis move up and down.

If it does then you could drop the laser down by a very small amount after a doing a certain amount of passes.
That will set the lasers point of focus further down inside the first pass or passes.

**Comment by Rick23n** [comment]
I understand what you are saying Stuart and there is a Z axis plug-in but I don't have a Z axis so I have to do it manually until or if I build one.

**Comment by StuartB4** [comment]
Oh ok Rick.

**Comment by Rick23n** [comment]
I have added a shroud to my laser so that I can put away the glasses. It now means that I don't have to lock the door to stop anyone coming into the room while the laser is on. I saw on another forum someone wanted to get rid of the glasses so maybe this will help them. I did NOT use the 450nm rated material ($300/sq ft) but I am assured that the 3mm dark green acrylic is safe. I have also added a red laser point to allow positioning of the laser when I need to have margin insets. The addition of an LED allows vision through the acrylic to show the workspace clearly.
![img_2432](https://user-images.githubusercontent.com/42868495/47076007-22010700-d241-11e8-9b28-82b3113551a0.JPG)
![img_2431](https://user-images.githubusercontent.com/42868495/47076040-2f1df600-d241-11e8-957a-03e6d18c63af.JPG)
![img_2434](https://user-images.githubusercontent.com/42868495/47076064-393ff480-d241-11e8-8363-7bd826c15487.JPG)
I have also ordered a Z axis to be added to my laser.
![z axis](https://user-images.githubusercontent.com/42868495/47076407-fb8f9b80-d241-11e8-974a-4172185c1df6.jpg)
 I am not sure yet how I am going to control it as grbl has the z axis in the config but the main control panel does not have z axis controls !(or have I missed something ?)
Also controling the number of passes is not capable in grbl. It would be nice to do this automatically for cutting.

**Comment by StuartB4** [comment]
Hi Rick.
I think you can use 2 Custom buttons for the Z Axis up and down.
I don't have a Z Axis but made 2 buttons for when I get one.
They seem to work when I change the Z in the GCode to X or Y so I assume they will work for Z.
I set the movement to 0.10 mm and when I press either button the axis moves a tiny amount.
Obviously changing that number will move it more / less depending on the setting.
I set Up as -0.10 and Down as 0.10
Like this:
G91 Z-0.10
and 
G91 Z0.10

![zaxis](https://user-images.githubusercontent.com/16980026/47083613-1396f900-d209-11e8-901c-a21a82a8a0d6.jpg)

**Comment by Rick23n** [comment]
Thank you for that Stuart, I am still learning but it makes sense. What were your thoughts on the multiple passes for cutting?

**Comment by StuartB4** [comment]
Hi Rick.
May be it's me but I don't quite understand what you mean by automatically for cutting.
I assume you know about the number of passes adjuster, at the right of the Prograss bar.
![untitled](https://user-images.githubusercontent.com/16980026/47099668-af3a6080-d22d-11e8-8197-ea45ec1d8809.jpg)

**Comment by Rick23n** [comment]
I need new glasses!!  my apologies I was thinking I had to do the extra passes manually by re-activating the G code

**Comment by StuartB4** [comment]
Thats ok Rick. Like you said, still learning.
I like to think we never stop.

**Comment by Rick23n** [comment]
Everything has been working great, then I decided to add a 'Z' axis. To do this I needed to upgrade my controller to accommodate the 3rd axis. I bought the controller (see picture) and the 'Z'axis from Banggood and set it up using the original 2 axis controller to test and after a few adjustments, that worked fine ( I used the X axis driver to test the new stepper). Now with the new controller I cannot turn the laser off or move the laser in any direction. I have used the jumpers supplied under the stepper motor drivers but am not sure if there needs to be any other bridges
![image](https://user-images.githubusercontent.com/42868495/48468657-3b917080-e838-11e8-9f1a-c2ef125c671a.png)

![image](https://user-images.githubusercontent.com/42868495/48469716-51ecfb80-e83b-11e8-9570-59eaca4e1485.png)
Stuart, you said you intended to add a Z axis at some time. Have you done this yet? It has cut down my print area substantially but I don't need it large at the moment. I can always add rails later.

**Comment by StuartB4** [comment]
NHi Rick.
No I have'nt added Z yet.

Are you sure you have the driver boards the correct way round.
This picture shows the current adjuster pot facing away from the motor leads, your's looks like it's facing towards them

![untitled](https://user-images.githubusercontent.com/16980026/48483660-f9ead080-e80b-11e8-88f7-17ed5d8c769d.jpg)

**Comment by StuartB4** [comment]
Wow this is confusing.
This image is from an item on Aliexpress.
It shows 2 pics of the sam 
![untitled](https://user-images.githubusercontent.com/16980026/48484112-2eab5780-e80d-11e8-87e8-6d264f470525.jpg)
board but the driver boards are different on both pics.

**Comment by StuartB4** [comment]
Rick. Are you sure the laser connects to the 3 pins you have yours on.
In the image above they have a jumper on 2 of the 3 pins where you have the laser plugged.

**Comment by Rick23n** [comment]
Slight problem-this board does not seem to have a TTL connection

**Comment by Rick23n** [comment]
Yes I see the jumper now and did realize that was why the laser could not be turned off - it is 12v power out but not controllable for the laser. I will try to connect to the laser Interface but that does not appear to have any voltage

**Comment by Rick23n** [comment]
With the jumper on the 12v I now have power to the stepper motors but will have to configure the Z axis motor the same as the others as the command you gave me does not work but when I had the Z axis connected to the X stepper it worked great on the old board. I still cant see the PWM pin for the TTL

**Comment by Rick23n** [comment]
Now with this $ command $J=G91Z20F500 the Z axis is moving correctly and the position of the stepper drivers are with adjuster pot facing the motor leads. Am still looking for the TTL connection. Cheers

**Comment by StuartB4** [comment]
I've just been looking at some more images of the board, I think you are right about it not having a PWM connection for the laser. 
The other way to do it is to change your setup completely.
If you use an Arduino UNO and a Protoneer CNC shield, but thats more messing about and more expense. You can use your existing motor driver boards though.
The shield is actually designed and sold from New Zealand, closer to you than me in the UK.
The website is really helpful as well, good support.
https://blog.protoneer.co.nz/arduino-cnc-shield


![untitled](https://user-images.githubusercontent.com/16980026/48490053-72598d80-e81c-11e8-8696-db6b3d1266da.jpg)

**Comment by Rick23n** [comment]
Thank you for that info, will re-look tomorrow 1.00am now Still can't work out how to power the laser off this board unless it requires fly leads from the 12v output to the connections shown as laser interface and high power ?

**Comment by Rick23n** [comment]
Hi Stuart. The 2 pin laser connector does not activate the laser when I set up a print job. The only way I have been able to get 12vdc to those pins is to run a negative rail from the 1st (-ve) of the 3 pins where 12v comes in from the power supply(where I had to put the jumper on pins 2/3)of course when I do that I have constant power to the laser and connecting the yellow TTL cable to pin 11 on the Nano does nothing. I have asked BIQU (the manufacturer) but all they said was "they don't have the files for it and besides the technician isn't here" (great!) I would like to get this controller up and running but may have to invest in the one from Protoneer as you suggested or find  another like this one that supports a laser up to 18W
![image](https://user-images.githubusercontent.com/42868495/48549274-0c076480-e91b-11e8-8641-f79e2fdcf2ec.png)

**Comment by StuartB4** [comment]
Thats strange Rick, 18 Watt ? the picture shows a TTL Laser Driver Board that can only handle  3 Watt Lasers.
I use that board myself for a 2 Watt Blue Laser.
I know it looks really messy but it works.

![img_5645](https://user-images.githubusercontent.com/16980026/48569847-c20e8680-e8fa-11e8-9e56-41e1d7733299.JPG)


![img_5647](https://user-images.githubusercontent.com/16980026/48569894-d94d7400-e8fa-11e8-9fd5-54c426547f5c.JPG)

**Comment by Rick23n** [comment]
Hi Stuart, these are the specs for the controller I posted 
https://www.ebay.com.au/itm/GRBL-Laser-controller-board-CNC-USB-3-Axis-Stepper-Motor-Driver-Controller-Board-/292652076251
GRBL Laser controller board CNC USB 3 Axis Stepper Motor Driver Controller Board for CNC Mini engraving machine
This is a GRBL open source code-based laser,  can be used for Laser Engraving machine, CNC Engraving Machine etc.
product specifics:
1.Application：use for DIY mini Engraving machine or Toy laser engraving machine
2.Applicable software：GRBL control、Un Gcode Sender、 GRBL controller，or other software that supports GRBL code
3.Communication:USB（USB-CH340）.
4.Power supply: voltage: 12V current about 4A , according to the load to increase or decrease the power supply.
5.Stepper motor driver: A4988 motor driver,16 microstep 1.5A phase current output
6.Support stepper motor: NEMA17 NEMA23 stepper motor, phase current less than 1.5A .
7.Laser interface: 12V or 5V, power less than 18W laser module.
8.Spindle motor interface: the maximum support 400W DC spindle motor.
9.With fan interface and TTL signal output interface (for controlling the laser module with TTL module).
10.Limit interface: 3-axis limit interface.
11.Probe interface: for Z axis tool zero.
12.Power interface: DC-005 2.0 interface
I guess you have to be careful when you don't know what you are doing!
Question : would it be possible to use my original board (2 axis controller which was working ok) WITH the new 3 axis controller somehow? I really don't want to do that, I would like to get the new one to work properly. The only thing not working on it is the laser will not fire up from the Arduino

**Comment by StuartB4** [comment]
Thats interesting Rick.
I noticed it says it has a TTL OUT. The bottom right socket.
If it has then that would come from the Arduino chip, the square one in the middle of the board with all the legs around it.
The other chip above it is the Communication:USB（USB-CH340）
If the TTL does come from that chip then I don't understand why it would need the extra driver board that's pictured with it.
Surely if it's a TTL out then you should just be able to plug the yellow wire from your laser driver board in to it to get the signal,  and plug your laser + and - in to the socket next to it.
The TTL out on that board is a 2 pin connector, the second pin is probably TTL - which you would probably have to ground, it may be connected to the - pin on the laser connector next to it.

I'd like to see it wired up to a machine but have looked around and can't find any more info on it anywhere. Looks like a nice board though.

**Comment by StuartB4** [comment]
Just found this, looks like it's made by or for Sainsmart.
They might have more info about it as they sell it on their website here:
https://www.sainsmart.com/products/3-axis-grbl-usb-driver-controller-board

Also they have a better image where you can see the TTL + and - markings on the board.
![untitled](https://user-images.githubusercontent.com/16980026/48624253-de1a3280-e9a3-11e8-873c-ca01d4a49c4d.jpg)

**Comment by StuartB4** [comment]
Yes Arduino built in.  It's the Atmel Mega328P chip.

**Comment by Rick23n** [comment]
OK Yes that board you posted appears to be the same board I'm talking about, certainly the write-up is identical even support up to 18w laser (Which you queried) I'm just about over this 3 axis one I have now absolutely no support from BIQU or Banggood, Banggood told me to find something on Utube!

**Comment by StuartB4** [comment]
Yeah that happens with a lot of stuff, can't wait to sell it you but then no support.
I've asked Sainsmart for some more info, just waiting for a reply, we shall see.

**Comment by StuartB4** [comment]
I noticed it comes with a CD as well.
I suppose that will be a bit of software for sending your images to the machine, something similar to LaserGRBL, and also the USB-CH340 Driver, which usually has to be installed first so it can communicate with your machine.

**Comment by Rick23n** [comment]
I have also tried connecting the Grd pin of the Arduino to the Neg of the laser, I thought maybe it would act the same as when I connected the 12v Neg to the laser and got full power but would only activate when the laser was printing, ALAS that doesn't work either

**Comment by StuartB4** [comment]
Hi Rick.
I received this today from Sainsmart. Handy but not a mention of how to use the TTL option or if it needs an external TTL board.

> Hey Stuart,
> 
> Thanks for reaching out!
> 
> Sorry that our product team is on business trip at present and we cannot answer your questions directly.
> 
> Here is the manual for this item:
> http://wiki.sainsmart.com/index.php/101-60-508
> 
> If you need any further help from us, please let me know.
> 
> Best regards,
> Lynn.

**Comment by Rick23n** [comment]
Hi Stuart, it does look as though you need to buy a TTL Module and the ones
that I just looked at cost more than the board. I wonder if they
(sainsmart) have them? I took 2 videos of my set-up showing Banggood that
the 3 axis controller was fine for the stepper motors but had no voltage at
the laser input when the system was running, they are going to credit my
account so that is good. I was about to order this new controller from
'louisa-shop' on ebay for $36.85 but I see that 'SainSmart's price is
$22.99. Now that we are not sure about the TTL I will have to do some more
research

**Comment by Rick23n** [comment]
I searched Sainsmart for a TTL laser module but it appears they don't have anything with that description and anywhere else comes up with just the laser. I've taken a punt and ordered it from Sainsmart so we shall find out one way or another. Strange but when you search their product range, this controller does not come up at all.

**Comment by StuartB4** [comment]
One other thing. I think the built in Arduino chip on that board might have an old version of GRBL installed in it or no GRBL at all.
Just because the link they supplied for the board manual also has links for Grbl_firmware_and_update, and also the USB Driver for the other chip.
When I looked at it it's version 0.9j. So the Arduino chip might still need upgrading to the version we are using now. Version 1.1.

**Comment by Rick23n** [comment]
Yes, good thought Stuart. If that board is any thing like the one that I bought, the FAN output is 12vdc constant (Not controlled by PWM)

**Comment by StuartB4** [comment]
Hi Rick.
Just had this reply from them.

> Hi Stuart,
> 
> Sorry for causing the confusion.
> 
> Since the FAN interface was connected to the 12V power supply directly, the positive and negative terminals of the laser module should be connected to the FAN. That is, the red (positive 12 volt) should be connected to the FAN+ port of the control board.
> 
> The Black (Negative/Gnd) should be connected to the FAN- and TTL- port because the TTL- and GND are not in the same GND and they should be connected.
> 
> If you used the laser port as the power source, the laser module will not work because the laser module needs the continuous power supply to control the laser intensity .
> 
> Best regards,
> Lynn

Bit easier to understand now.

**Comment by StuartB4** [comment]
Hi Lynn.

Thanks for your time and explanation, you have been really helpful.

That makes it a lot easier to understand.

Kind regards

Stuart.

**Comment by StuartB4** [comment]
Sorry Rick, I think I just sent you a reply meant for Lynn by mistake.
It is probably in your emails.

**Comment by StuartB4** [comment]
Like this.

![untitled](https://user-images.githubusercontent.com/16980026/48844731-92e09500-ed92-11e8-8ac8-449d9d0538b3.jpg)

**Comment by Rick23n** [comment]
The new board arrived and after much translation I have it working to a degree. The new software is certainly not as manageable as grbl. Have only been able to vector so far and cannot find the Z axis control. I did have control of the Z axis when I was using the new board with grbl but the new software doesn't show it. Maybe it has something to do with the CNC settings when I get there.

**Comment by StuartB4** [comment]
Hi Rick.
What software is it?

**Comment by Rick23n** [comment]
Hi Stuart
Well firstly this is the link for the board that I purchased:
https://picclick.ca/GRBL-Laser-controller-board-CNC-USB-3-Axis-312323208146.html
It came with a mini CD that had the "instructions" but no name for the software. See below:
laser controller Software instructions
1. software features introduced：
●Direct engraving images; No need to generate engraving code. Click carving to load
pictures.
●Direct engraving test; No need to generate engraving code. Enter text to click
carving.
●Support G code; Direct carving engraving file by third party software
●Laser power regulation; 1000 grade laser power regulation
●Weak light positioning; Click weak light modulation to determine the starting point
of the laser
●Support engraving graphics Preview; Preview to ensure the accuracy of the
engraving position
●Support any set starting point; Set laser starting point according to material shape
Support single line text carving; super small text can be carved
●Support dot matrix text carving; commonly used in engraving production date and
other information
●Contour engraving; Hollow effect, commonly used in laser cutting
●Support solid engraving
●Support grayscale carving
●Automatic repeat carving, suitable for solid material at one cutting.
2. software installation：
Click on the "Laser Controller.exe" software installation, only prompted to
complete, complete the software installation.
3. Software operation：
Click the desktop icon "Laser Controller" to open the software, the interface as
shown below:
![image](https://user-images.githubusercontent.com/42868495/49644709-f17d6280-fa64-11e8-8841-ca9227f3b11a.png)
4. language selection：
Click the upper right corner of the software, select the language,
currently supports English and Chinese 2 languages.
![image](https://user-images.githubusercontent.com/42868495/49644822-6355ac00-fa65-11e8-8cf7-896133b30335.png)
After all that it doesn't show Z axis control. 90% of it is in Chinese or reverts to Chinese after you change one of the parameters. The log is only in Chinese. It (compared to GRBL) has only 1 greyscale setting. You have to continually recognize the port (because it reverts to Chinese) You have to check each time you go to print that it is 'true' & not back to front or upside down.
So what I have found is that this new board works perfectly with Laser GRBL after I flashed the newer 1.1 version - AND I have Z axis control ! and yes the PWM function was only available after bridging the Neg of the TTL to Neg of the Laser, they are not automatically connected as some have suggested. 
The few burns that I have done are a marked improvement over the original controller supplied with the laser which I had upgraded from Benbox to Laser GRBL. This new board appears to be far more flexible in terms of power and speed, but I have a long way to go yet to fully understand the options and limits.
Merry Christmas

**Comment by StuartB4** [comment]
Hi Rick.
Wow, that software seems a bit of a nightmare.
Glad you have sorted the Z axis control out. 
Also happy to know you could upgrade GRBL to 1.1, if they don't tell you it's possible then it's all trial and error.
Good to know the TTL/PWM works as they said at Sainsmart.
As you say you still have a long way to go, at least with the hardware sorted you can concentrate more on the actual creative side.

You have a great Christmas too.

**Comment by StuartB4** [comment]
Hi Rick.
I have limited knowledge when it comes to CNC as a Router.
As far as the controller board is concerned it looks pretty straight forward.

I don't know what power supply you have at the minute, but it looks like you might need an extra one.
12 Volt, at least 4 Amps for the Spindle motor. Connected to the Bottom 2 holes of the green connector.
The spindle motor wires go in to the top 2 of the same connector.

Swap the Laser / CNC option jumper just underneath the blue relay to CNC.
I assume the relay switches the spindle motor on.

The probe (optional) for sensing the height of the spindle motor cutting tool from workpiece is connected to the last 2 pins on the right of the X/Y/Z/Probe input pins.
Usually that it just 2 wires, one with a metal plate on the end and one with a crocodile clip on it.
You place the one with the metal plate on to the worktop underneath the cutting tool and clip the crocodile clip to the actual cutting tool.
Then you use the Probe option in whatever software you use, that will move the Z axis spindle motor up and down to position it at the correct height for the workpiece.

I would'nt like to say which software to use as there are so many options.

![htb](https://user-images.githubusercontent.com/16980026/52055753-318a2e80-2558-11e9-9a02-4bc89b5b6cb3.jpg)

**Comment by Rick23n** [comment]
Sorry Stuart when I read your message as an email it did not show your
reference to UGS. It was not until I checked the forum that I saw that
Cheers Rick

**Comment by Rick23n** [comment]
Yes Stuart it will be interesting. The power supply that currently runs my 16 watt laser is 12v 4amp so hopefully it will handle the requirements of the CNC. Although the 12v is a bit of a worry, I see some that are 48v.  I am assuming that the grbl software will control the router once the parameters have been set in $30 & $31. All the other settings would be the same as for the  laser (I assume) I will have to do a little more research in regard to the probe pins tho. I have done some engraving into acrylic but the burn is not deep enough without leaving an uneven surface finish I will need the router to give me a depth of up to 0.5 of a mill that can be filled with paint to give a pattern in the acrylic which will be back-lit. The size of the finished product will be 500mm x 2000mm hence the upgrade to my small laser carriage.
